Backend/app/dashboard/Lib/src/Tools/Auth/Common.hs

```diff
@@ -37,27 +39,33 @@ type AuthFlow m r =
 verifyPerson ::
   (AuthFlow m r, Redis.HedisFlow m r) =>
   RegToken ->
-  m (Id DP.Person, Id DMerchant.Merchant)
+  m (Id DP.Person, Id DMerchant.Merchant, City.City)
 verifyPerson token = do
   key <- authTokenCacheKey token
   authTokenCacheExpiry <- getSeconds <$> asks (.authTokenCacheExpiry)
   mbTuple <- getKeyRedis key
-  (personId, merchantId) <- case mbTuple of
-    Just (personId, merchantId) -> return (personId, merchantId)
+  (personId, merchantId, city) <- case mbTuple of
```

Comment:
Backward compatibility.

Reply:
Fixed it, by first decoding the value in `Text` and using `readMaybe` twice, for both cases. Please check if something can go wrong.